microcosm_pubsub/conventions.py

```diff
@@ -0,0 +1,98 @@
+"""
+Convention-driven pubsub.
+
+"""
+from enum import Enum, unique
+
+from inflection import underscore
+from marshmallow import fields
+
+from microcosm_pubsub.codecs import PubSubMessageSchema
+
+
+@unique
+class LifecycleChange(Enum):
+    """
+    CRUD lifecycle changes that may be announced via pubsub.
+
+    Note that `Changed` is strongly discouraged; use immutable resources instead.
+
+    """
+    Changed = u"changed"
+    Created = u"created"
+    Deleted = u"deleted"
+
+    def matches(self, media_type):
+        return self.value in media_type.split(".")
+
+
+def make_media_type(resource, lifecycle_change=None, organization=None, public=False):
+    """
+    Generate a canonical media type for a message that announces a lifecycle change to a resource.
+
+    :param resource: the resource (name)
+    :param lifecycle_change: the change that happened to the resource
+    :param organization: the vendor organization to which this resource belongs
+    :param public: whether the resource is meant for public/external consumption
+
+    """
+    if lifecycle_change is None:
+        lifecycle_change = LifecycleChange.Created
+    if organization is None:
+        organization = "globality"
+
+    return "application/vnd.{}.{}.{}.{}".format(
+        # use a vendor specific media type
+        organization,
+        # differentiate messages sent over a pubsub channel from other messages
+        "pubsub" if public else "pubsub._",
+        # qualify the message with the kind of lifecycle change
+        lifecycle_change.value,
+        # specify the resource name
+        underscore(resource),
```

Comment:
At the moment, it's assumed to be the resource name. I was contemplating duplicating [this function](https://github.com/globality-corp/microcosm-flask/blob/develop/microcosm_flask/naming.py#L11) so we can pass in classes as well.